statsmodels/tsa/holtwinters.py

```diff
@@ -0,0 +1,821 @@
+
+#(A,A) (Ad,A)
+
+
+def _holt_win_add_add_dam(x, xi, p, y, l, b, s, m, n, max_seen):
+    alpha, beta, gamma, phi, alphac, betac, gammac, y_alpha, y_gamma = _holt_win_init(
+        x, xi, p, y, l, b, s, m)
+    if alpha * beta == 0.0:
+        return max_seen
+    if beta > alpha or gamma > 1 - alpha:
+        return max_seen
+    for i in range(1, n):
+        l[i] = (y_alpha[i - 1]) - (alpha * s[i - 1]) + \
+            (alphac * (l[i - 1] + phi * b[i - 1]))
+        b[i] = (beta * (l[i] - l[i - 1])) + (betac * phi * b[i - 1])
+        s[i + m - 1] = y_gamma[i - 1] - \
+            (gamma * (l[i - 1] + phi * b[i - 1])) + (gammac * s[i - 1])
+    return sqeuclidean((l + phi * b) + s[:-(m - 1)], y)
+
+#(M,A) (M,Ad)
+
+
+def _holt_win_mul_add_dam(x, xi, p, y, l, b, s, m, n, max_seen):
+    alpha, beta, gamma, phi, alphac, betac, gammac, y_alpha, y_gamma = _holt_win_init(
+        x, xi, p, y, l, b, s, m)
+    if alpha * beta == 0.0:
+        return max_seen
+    if beta > alpha or gamma > 1 - alpha:
+        return max_seen
+    for i in range(1, n):
+        l[i] = (y_alpha[i - 1]) - (alpha * s[i - 1]) + \
+            (alphac * (l[i - 1] * b[i - 1]**phi))
+        b[i] = (beta * (l[i] / l[i - 1])) + (betac * b[i - 1]**phi)
+        s[i + m - 1] = y_gamma[i - 1] - \
+            (gamma * (l[i - 1] * b[i - 1]**phi)) + (gammac * s[i - 1])
+    return sqeuclidean((l * phi * b) + s[:-(m - 1)], y)
+
+
+class HoltWintersResults(base.Results):
+    def __init__(self, model, params, **kwds):
+        self.data = model.data
+        super(HoltWintersResults, self).__init__(model, params, **kwds)
+
+    def predict(self, start=None, end=None):
+        return self.model.predict(self.params, start, end)
+
+    def forecast(self, steps=1):
+        return self.model._predict(h=steps, **self.params).fcast
+
+
+class HoltWintersResultsWrapper(wrap.ResultsWrapper):
+    _attrs = {'fittedvalues': 'rows',
+              'level': 'rows',
+              'resid': 'rows',
+              'season': 'rows',
+              'slope': 'rows'
+              }
+    _wrap_attrs = wrap.union_dicts(wrap.ResultsWrapper._wrap_attrs,
+                                   _attrs)
+    _methods = {'predict': 'dates'}
+    _wrap_methods = wrap.union_dicts(wrap.ResultsWrapper._wrap_methods,
+                                     _methods)
+
+
+wrap.populate_wrapper(HoltWintersResultsWrapper, HoltWintersResults)
+
+
+class ExponentialSmoothing(tsbase.TimeSeriesModel):
+    """
+    Holt Winter's Exponential Smoothing
+
+    Parameters
+    ----------
+    endog : array-like
+        Time series
+    trend : {"add", "mul", None}, optional
+        Type of trend component.
+    damped : bool, optional
+        Should the trend component be damped.
+    seasonal : {"add", "mul", None}, optional
+        Type of seasonal component.
+    m : int, optional
+        The number of seasons to consider for the holt winters.
+    """
+
+    def __init__(
```

Comment:
Normal style here is something like 

```
def __init__(self, endog, trend=None, damped=False, seasonal=None, m=None, 
             dates=None, freq=None, missing='none', **kwargs):
```

This is true in a few places below.

Reply:
That was what that autopep8 command did.
--aggressive maybe tone it down a bit leave that off?